x-pack/plugins/alerting/server/authorization/alerting_authorization_kuery.ts

```diff
@@ -52,6 +52,10 @@ export function asFiltersByRuleTypeAndConsumer(
         ),
       ];
 
+      while (andNodes.length && andNodes[andNodes.length - 1] == null) {
+        andNodes.pop();
+      }
```

Comment:
Why are we only removing from the end?
Wouldn't it be better (and more readable) if we avoid adding the condition if there are no authorised consumers? 🤔

Reply:
Improved in commit dedbb75dba432f7a53f44b583901f62ba13c5beb